nx/guides/modoc_agg_nx_01.livemd

```diff
@@ -0,0 +1,826 @@
+\end{bmatrix}
+
+\end{bmatrix}
+\end{bmatrix}
+
+
+
+\begin{bmatrix} x=0
+  \begin{bmatrix} y=0
+
+```
+\begin{bmatrix}
+  & 1, 2, 3, & \\
+  & 1, -2, 3, &
+\end{bmatrix}
+\\ 
+y=1
+\begin{bmatrix} 
+  & 4, -3, 2 & \\
+  & -4, 3, 2 &
+\end{bmatrix}
+\\ 
+y=2
+\begin{bmatrix}
+  & 5, -1, 3 &\\
+  & -5, 1, 3 &
+\end{bmatrix}
+```
+
+\end{bmatrix}
+  \\
+  x=1
+  \begin{bmatrix} y=0
+
+```
+\begin{bmatrix}
+  & 4, 6, 1 & \\
+  & 4, -6, 1 &
+\end{bmatrix}
+\\ 
+y=1
+\begin{bmatrix}
+  & 1, 2, 3 & \\
+  & 1, -2, 3 &
+\end{bmatrix}
+\\ 
+y=2
+\begin{bmatrix}
+  & 4, -3, 2 & \\
+  & -4, 3, 2 &
+\end{bmatrix}
+```
+
+\end{bmatrix}
+\end{bmatrix}
+$$
+
+<!-- livebook:{"break_markdown":true} -->
+
+Firstly, let's check some characteristics of this tensor: they use full tensor aggregation functions.
+
+```elixir
+%{
+  dimensions: %{
+    x: Nx.axis_size(t, :x),
+    y: Nx.axis_size(t, :y),
+    z: Nx.axis_size(t, :z),
+    t: Nx.axis_size(t, :t)
+  },
+  n: Nx.size(t),
+  shape: Nx.shape(t),
+  rank: Nx.rank(t),
+  analysis: %{
+    most_frequent_nb: Nx.mode(t) |> Nx.to_number(),
+    smallest_element: %{
+      value: Nx.reduce_min(t) |> Nx.to_number(),
+      position: Nx.argmin(t) |> Nx.to_number()
+    },
+    greatest_element: %{
+      value: Nx.reduce_max(t) |> Nx.to_number(),
+      position: Nx.argmax(t) |> Nx.to_number()
+    },
+    no_zero_nb: Nx.all(t) |> Nx.to_number()
+  }
+}
+```
+
+#### Single row aggregation, along an axis
+
+> ❗ We are going to use the key `:axis` below. The following functions `argmin`, `argmax`, `median` and `mode` use `:axis: <symbol> or number` in singular mode. All of the other aggregating functions use `axes: [<symbol> or number]`; these have multi-axis aggregation implemented.
+
+When you aggregate along an axis, you are going to reshape the tensor and to aggregate with a function, in other words perform a reduction. It is important to consider the _ordering_ of the axes.
+
+<!-- livebook:{"break_markdown":true} -->
+
+###### Aggregate along the first axis `axis: :x` (equivalently `axis: 0`).
+
+The shape of our original tensor  `t`  is `[x: 2, y: 3, z: 2, t: 3]`. When we aggregate along the axis `:x`, it collapses and the shape of the resultant tensor `r` is `[y: 3,z: 2,t: 3]`. The rule is that every axis before the selected one will remain in the structure of the tensor as well as every axis after the selected one.
+
+How? You collect from each slice of `x` the elements that have the _same_ remaining indexes "on the right" $y,z,t$. In the case of our tensor, you have 2 `x`-slices so you build a sublist of 2 elements, and then apply the aggregating function on it. This gives you the value of the resulting tensor $r$ at the location $r[y][z][t]$:
+
+$$
+\rm{agg}\big(t[0][y][z][t], t[1][y][z][t]\big) = r[y][z][t]
+$$
+
+The aggregating function used below is `argmin`: the index of the smallest element of the  sublist. In case of several occurences, the `:tie_break` attribute takes the lowest index.
```

Comment:
```suggestion
The aggregation function used below is `argmin`: the index of the smallest element of the sublist. In case of several occurrences, the `:tie_break` option takes the lowest index by default.
```

Reply:
Again, I think it would be clearer to the reader if Nx slicing notation is used below instead of Latex rendering